src/AtomsMCMoves.jl

```diff
@@ -0,0 +1,368 @@
+"""
+Module containing functions for performing Monte Carlo moves on atomic/molecular systems.
+"""
+module AtomsMCMoves
+
+using ExtXYZ
+using AtomsBase
+using Setfield
+using Distributions
+using Unitful
+using StaticArrays
+# using LinearAlgebra
+
+using ..Potentials
+using ..AbstractWalkers
+using ..EnergyEval
+
+export periodic_boundary_wrap!
+export MC_random_walk!, MC_nve_walk!
+
+
+"""
+    periodic_boundary_wrap!(pos::SVector{3,T}, system::AbstractSystem) where T
+
+Wrap the position vector `pos` according to the periodic boundary conditions of the `system`. If the
+boundary condition is `Periodic()`, the position is wrapped using the modulo operator. If the boundary
+condition is `DirichletZero()`, the position is wrapped by reflecting the position vector across the boundary.
+
+# Arguments
+- `pos::SVector{3,T}`: The position vector to be wrapped.
+- `system::AbstractSystem`: The system containing the periodic boundary conditions.
+
+# Returns
+The wrapped position vector.
+
+"""
+function periodic_boundary_wrap!(pos::SVector{3,T}, system::AbstractSystem) where T
+    pbc = system.boundary_conditions
+    box = system.bounding_box
+    new_pos = Vector{typeof(0.0u"Å")}(undef, 3)
+    for i in eachindex(pbc)
+        if pbc[i] == Periodic() # wrap the position
+            new_pos[i] = mod(pos[i], box[i][i])
+        elseif pbc[i] == DirichletZero() # reflect the position
+            if pos[i] > box[i][i]
+                new_pos[i] = box[i][i]*2 - pos[i]
+            else
+                new_pos[i] = pos[i]
+            end
+        else
+            error("Unsupported boundary condition: $(pbc[i]).")
+        end
+    end
+    pos = SVector{3,T}(new_pos)
+    return pos
+end
+
+"""
+    mean_sq_displacement(at::AtomWalker, at_orig::AtomWalker)
+
+Calculate the mean squared displacement before and after random walk(s).
```

Comment:
Something to think about @yangmr04: If the user is interested in mean displacements, not restricting the particle coordinates might be preferable.